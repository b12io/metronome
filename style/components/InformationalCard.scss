.informational-card {
  position: relative;
  border: $ds-card-border;
  border-radius: $ds-input-border-radius;
  box-shadow: $ds-box-shadow-card;
  padding: spacing(xxsm);
  font-size: $display-xsmall-font-size;
  line-height: 20px;

  &__before {
    position: absolute;
    height: 7px;
    top: -1px;
    left: -1px;
    right: -1px;
    border-top-left-radius: $ds-input-border-radius;
    border-top-right-radius: $ds-input-border-radius;
    border: 1px solid rgba(0, 0, 0, 0.08);
    border-bottom: none;
  }

  &__heading {
    display: flex;
    font-weight: $ds-font-weight-bold;
    padding-bottom: spacing(xxxsm);
  }

  &__subheading {
    padding-left: spacing(xsm);
    color: $ds-body-light-color;
  }

  &__action-item {
    padding: spacing(xxxsm) 0 0 spacing(xsm);
  }
}
