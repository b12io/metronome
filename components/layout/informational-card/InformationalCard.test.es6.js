/* global expect */
import React from 'react'
import renderer from 'react-test-renderer'

import InformationalCard from './InformationalCard.es6.js'

describe('InformationalCard component', () => {
  let testProps

  beforeEach(() => {
    testProps = {
      heading: 'Data.',
      subheading: 'Cookies.'
    }
  })

  test('should render correctly with children', () => {
    const tree = renderer.create(
      <InformationalCard {...testProps}>
        <div>Test children prop</div>
      </InformationalCard>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('should render correctly without children', () => {
    const tree = renderer.create(<InformationalCard {...testProps} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
