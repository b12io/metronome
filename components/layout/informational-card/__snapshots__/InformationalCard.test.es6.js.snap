// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InformationalCard component should render correctly with children 1`] = `
<div
  className="informational-card"
>
  <span
    className="informational-card__before"
    style={
      Object {
        "background": "#F5BE33",
      }
    }
  />
  <div
    className="informational-card__heading"
  >
    <svg
      className=""
      height={16}
      style={
        Object {
          "marginRight": 5,
          "marginTop": 2,
          "minWidth": 16,
        }
      }
      viewBox="0 0 16 16"
      width={16}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8 0C3.6 0 0 3.6 0 8s3.6 8 8 8 8-3.6 8-8-3.6-8-8-8zm1 12H7V7h2v5zM8 6c-.6 0-1-.4-1-1s.4-1 1-1 1 .4 1 1-.4 1-1 1z"
        fill="#F5BE33"
        fillRule="nonzero"
      />
    </svg>
    Data.
  </div>
  <div
    className="informational-card__subheading"
  >
    Cookies.
  </div>
  <div
    className="informational-card__action-item"
  >
    <div>
      Test children prop
    </div>
  </div>
</div>
`;

exports[`InformationalCard component should render correctly without children 1`] = `
<div
  className="informational-card"
>
  <span
    className="informational-card__before"
    style={
      Object {
        "background": "#F5BE33",
      }
    }
  />
  <div
    className="informational-card__heading"
  >
    <svg
      className=""
      height={16}
      style={
        Object {
          "marginRight": 5,
          "marginTop": 2,
          "minWidth": 16,
        }
      }
      viewBox="0 0 16 16"
      width={16}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8 0C3.6 0 0 3.6 0 8s3.6 8 8 8 8-3.6 8-8-3.6-8-8-8zm1 12H7V7h2v5zM8 6c-.6 0-1-.4-1-1s.4-1 1-1 1 .4 1 1-.4 1-1 1z"
        fill="#F5BE33"
        fillRule="nonzero"
      />
    </svg>
    Data.
  </div>
  <div
    className="informational-card__subheading"
  >
    Cookies.
  </div>
</div>
`;
